assignment.rb

```diff
@@ -0,0 +1,100 @@
+require "csv"
+require "./user.rb"
+
+### CSV set-up
+current_users = []
+csv = CSV.foreach('./users.csv', headers: true, header_converters: :symbol) { |row|
+  row.to_hash
+  user = User.new(row[:name], row[:pin], row[:balance])
+# Curious if there is a way to do this without creating an array.
+# All the User info is right here, I just don't know how to access it...
+# ... unless I push it to an array.
+  current_users.push(user)
+}
+
+## ATM functionality
+
+def user_input(text)
+  var = ""
+  while var == ""
+    puts text
+    print "> "
+    var = gets.chomp
+  end
+  var
+end
+
+def error_msg(text)
+  puts text
+end
+# Method to start back at the sign in menu, but I can't figure it out yet.
+# def start_over
+#   yes_no = nil
+#   puts "Would you like to start over? Please enter either y/n."
+#   yes_no = gets.slice(0)
+#   if yes_no == "y" || "Y"
+#
+#   elsif yes_no == "n" || "N"
+#     puts "Thank you. Have a good day!"
+#   end
+# end
+
+### Login
+
+current_name = user_input("Please enter your first name:")
+current_pin = user_input("Please enter your PIN:")
+
+verified_user = nil
+current_users.each do |user|
+  if user.name == current_name && user.pin == current_pin
+  verified_user = user
```

Comment:
Syntax & spacing in programs is important not only for the code to run correctly but for code readability, as well. This line should be indented another 2 spaces, for it is nested within the `if` block